js/ui/appDisplay.js

```diff
@@ -1589,10 +1598,11 @@ var AppSearchProvider = new Lang.Class({
                 let isOnDesktop = this._iconGridLayout.hasIcon(appID);
 
                 // exclude links that are not part of the desktop grid
-                if (!(app && app.should_show() && !(isLink && !isOnDesktop)))
+                if (!(app && parentalControlsManager.shouldShowApp(app) &&
+                    !(isLink && !isOnDesktop)))
```

Comment:
Nitpick: I guess it would be better here to

```javascript
if (!app ||
    !parentalControlsManager.shouldShowApp(app) ||
    (isLink && !isOnDesktop)) {
    return false;
}
```